src/mono/mono.proj

```diff
@@ -866,7 +866,12 @@
     </ItemGroup>
     <!-- OSX specific options -->
     <ItemGroup Condition="'$(TargetsOSX)' == 'true'">
-      <_MonoCMakeArgs Include="-DCMAKE_OSX_DEPLOYMENT_TARGET=$(macOSVersionMin)" />
+      <_MonoCMakeArgs Condition="'$(Platform)' != 'arm64'" Include="-DCMAKE_OSX_DEPLOYMENT_TARGET=$(macOSVersionMin)" />
```

Comment:
I assume we need to set a different min version because arm64 requires >= 11.0?
We should rather set a different value for `macOSVersionMin` in src/mono/Directory.Build.props in that case.